docs/typescript.md

```diff
@@ -0,0 +1,93 @@
+# Typescript
+
+## General considerations
+
+As we are progressively migrating the entire codebase to TS, we will have a mix of `.js` and `.ts` files in the process. For this reason and to make the migration easier for beginners, we have disabled the [`noImplicitAny`](https://www.typescriptlang.org/tsconfig#noImplicitAny) option from our `tsconfig.json` which means that you can use the `any` type and you will not get an error from that.
+When we reach the moment where the entire codebase is in TS, we will enable `noImplicitAny` and increase the checker’s strictness.
+
+This is just starting, any suggestion is always welcomed and appreciated.
+
+## Migration guide & contribution opportunity 🎉
+
```

Comment:
I'd add a quick blurb here - "we'd love your help in migrating our codebase!"?

We should specifically point people the issue as a call to action: #6392